### PR TITLE
 Add version details.

### PR DESCRIPTION
**Pull request by gillessed**
Add a versionDetails() function which returns an object
containing the latest tag name and the commit distance to the tag.

This is useful information for publishing to almanac.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -3,10 +3,14 @@ Git-Version Gradle Plugin
...
-runs the JGit equivalent of `git describe` to determine a version string. It behaves exactly 
-as the JGit `git describe` method behaves, except that when the repository is in a dirty 
-state, appends `.dirty` to the version string.
+When applied, Git-Version adds two methods to the target project.
+
+The first, called `gitVersion()`, runs the JGit equivalent of `git describe` to determine a version string.
+It behaves exactly as the JGit `git describe` method behaves, except that when the repository is in a dirty 
+state, appends `.dirty` to the version string. 
+
+The second, called `versionDetails()`, return an object containing the specific details of the version string:
```
Comment (markelliot) [inline readme.md]: "returnS"

File: readme.md [hunk 2]
```diff
@@ -20,6 +24,12 @@ Set the version of a project by calling:
 
     version gitVersion()
 
+You can get an object containing more detailed information by calling:
+
+    details = versionDetails()
```
Comment (markelliot) [inline readme.md]: `def details =`

File: src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy [hunk 3]
```diff
@@ -42,12 +40,42 @@ class GitVersionPlugin implements Plugin<Project> {
             }
         }
 
+        project.ext.versionDetails = {
+            File gitDir = getRootGitDir(project.rootDir)
+
+            try {
+                Git git = Git.wrap(new FileRepository(gitDir))
+                String description = git.describe().call();
```
Comment (markelliot) [inline src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy:48]: for consistency, this really ought to call the same thing as gitVersion()

File: src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy [hunk 4]
```diff
@@ -42,12 +40,42 @@ class GitVersionPlugin implements Plugin<Project> {
...
+
+                // Description has no git hash so it is just the tag name
+                if(!(description =~ /.*g.?[0-9a-fA-F]{3,}/)) {
+                  return new VersionDetails(description, 0);
+                }
+
+                String[] split = description.split('-')
```
Comment (markelliot) [inline src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy]: I think it'd be cleaner here to use a regular expression then to split things up and sow them back together.

**Comment by markelliot** [comment]
Looks reasonable, couple of nits + would like to see some tests.

File: src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy (hunk 3 above)
Comment (gillessed) [inline src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy:48]: Don't they both call `git.describe()` already?

Comment (markelliot) [inline src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy:48]: It'd be good to encapsulate the calls. I've been meaning to clean this up to memoize anyway, can leave as-is and I'll come back to it later.

**Comment by markelliot** [comment]
looks good to me, squash?

**Pull request merged**